tests/rptest/redpanda_cloud_tests/high_throughput_test.py

```diff
@@ -881,6 +886,172 @@ def _wait_for_traffic(self, producer, acked, timeout=1800):
                 f"Low throughput while preparing for the test: {_throughput} MB/s"
             ) from None
 
+    # returns a tuple of (replicas, readyReplicas)
+    def get_cluster_replicas(self, cluster_name: str) -> tuple[int, int]:
+        if self.operator_version == 2:
+            return self._get_cluster_replicas_v2(
+            ), self._get_cluster_ready_replicas_v2()
+        return self._get_cluster_replicas(
+            cluster_name), self._get_cluster_ready_replicas(cluster_name)
+
+    # sets new target replica count for the cluster
+    def set_cluster_replicas(self, cluster_name: str, new_replicas: int):
+        # for operator V2, we scale the statefulset because that seems
+        # to be the only way that works
+        if self.operator_version == 2:
+            self.redpanda.kubectl.cmd([
+                'scale', '-n=redpanda', 'statefulset', 'redpanda-broker',
+                f'--replicas={new_replicas}'
+            ])
+            return
+
+        # for operator v1, we patch the cluster CRD instead
+        patch = [{
+            'op': 'replace',
+            'path': '/spec/replicas',
+            'value': new_replicas
+        }]
+        patch_str = json.dumps(patch)
+
+        # kubectl patch -n=redpanda cluster rp-clkd0n22nfn1jf7vd9t0 --type=json -p='[{"op":"replace","path":"/spec/replicas","value":4}]'
+        self.redpanda.kubectl.cmd([
+            'patch', '-n=redpanda', 'cluster', cluster_name, '--type=json',
+            f"-p='{patch_str}'"
+        ])
+
+    def _kube_autoretry_int(self,
+                            command: list[str],
+                            max_retries: int = 10) -> int:
+        result = ''
+        retry_count = 0
+        while retry_count < max_retries:
+            result = self.redpanda.kubectl.cmd(command)
+            if type(result) is str and result != '':
```

Comment:
this is how we check for "command failed"? it looks very strange

Reply:
(it should be based on the return value)

Reply:
We have seen repeated failures where `self.redpanda.kubectl.cmd` successfully returns an empty string when trying to fetch these specific fields, without any error or different return value. This function is specifically intended to work around that case. (I'll add some documentation about this above the function.)

Reply:
Documentation has been added, with a note to check DEVPROD-2135.

Reply:
Right, I'm suggesting that KubectlTool.cmd() or whatever is implemented in a very strange way which should be fixed.